src/NLog/Internal/ObjectReflectionCache.cs

```diff
@@ -42,15 +42,23 @@ namespace NLog.Internal
     using System.Linq;
     using System.Reflection;
     using NLog.Common;
-    using NLog.Internal;
+    using NLog.Config;
 
     /// <summary>
     /// Converts object into a List of property-names and -values using reflection
     /// </summary>
-    internal class ObjectReflectionCache
+    internal class ObjectReflectionCache : IObjectTypeTransform
     {
         readonly MruCache<Type, ObjectPropertyInfos> _objectTypeCache = new MruCache<Type, ObjectPropertyInfos>(10000);
-        static readonly Dictionary<Type, Func<object, IDictionary<string,object>>> _objectTypeOverride = new Dictionary<Type, Func<object, IDictionary<string, object>>>();
+        private IObjectTypeTransform ObjectTypeReflection => _objectTypeReflection ?? (_objectTypeReflection = ConfigurationItemFactory.Default.ObjectTypeTransform);
```

Comment:
I would be nice if we could decouple this from ConfigurationItemFactory.Default, but I guess that's a lot more work :(

Reply:
Until I have completed my refactoring of NLog 5.0 dependency injection, then ConfigurationItemFactory is currently the goto-place for dependency injection stuff.